index.js

```diff
@@ -320,13 +325,93 @@ module.exports = {
     return options;
   },
 
+  _shouldHandleTypeScript(config) {
+      let emberCLIBabelConfig = config['ember-cli-babel'] || {};
+      if (typeof emberCLIBabelConfig.enableTypeScriptTransforms === 'boolean') {
+        return emberCLIBabelConfig.enableTypeScriptTransforms;
+      }
+      let typeScriptAddon = this.parent.addons
+        && this.parent.addons.find(a => a.name === 'ember-cli-typescript');
+      return typeof typeScriptAddon !== 'undefined'
+        && semver.gte(typeScriptAddon.pkg.version, '4.0.0-alpha.0');
+  },
+
+	_buildClassFeaturePluginConstraints(constraints, config) {
+    // With older versions of ember-cli-typescript, class feature plugins like
+    // @babel/plugin-proposal-class-properties were run before the TS transform.
+    if (!this._shouldHandleTypeScript(config)) {
+      constraints.before = constraints.before || [];
+      constraints.before.push('@babel/plugin-transform-typescript');
+    }
+
+    return constraints;
+  },
+
+  _addTypeScriptPlugins(plugins) {
+    const { hasPlugin, addPlugin } = require('ember-cli-babel-plugin-helpers');
+
+    if (hasPlugin(plugins, '@babel/plugin-proposal-optional-chaining')) {
+      if (this.parent === this.project) {
+        this.project.ui.writeWarnLine(`${
+          this._parentName()
+        } has added the optional chaining plugin to its build, but ember-cli-babel provides this by default now when ember-cli-typescript >= 4.0 and typescript >= 3.7 are installed! You can remove the transform, or the addon that provided it.`);
```

Comment:
Just to double check, this is added by default because TypeScript requires it? Also, how do you know that typescript >= 3.7 is being used (does e-c-ts bring a minimum of typescript@3.7?)?

Reply:
Side note, we may want to make a more official policy to include things like this in the future (e.g. "stage-3 proposals are added by default"), not sure but that would _probably_ require an RFC. 

What do you think @pzuraq?

Reply:
Yes, this is because these features are part of TS >= 3.7. The reason these have to be added unconditionally is because we *don't* know the TypeScript version an addon might have used.